### DISCUSSION TITLE
[volume.check.disk] repair only not deleted keys option

### DISCUSSION THREAD

**Discussion by kmlebedev** [seaweedfs/seaweedfs#3311] [Ideas]
@chrislusf  Perhaps it's not expensive to check that the key has not been deleted?
I also want to repair keys only older than 1 minute

**Comment by chrislusf** [comment]
After `readSourceNeedleBlob` and before `writeNeedleBlobToTarget` in command_volume_check_disk.go, the needle blob can be convert into a needle object similar to this in `needle_read.go:95`
```
n.ReadBytes(bytes, offset, size, version)
```

So you can get the `n.AppendAtNs` timestamp. 

Hope this is useful.

**Reply by kmlebedev** [reply]
done https://github.com/seaweedfs/seaweedfs/pull/3909
